Library/Homebrew/extend/os/linux/hardware/cpu.rb

```diff
@@ -20,12 +20,24 @@ def family
         # See https://software.intel.com/en-us/articles/intel-architecture-and-processor-identification-with-cpuid-model-and-family-numbers
         # and https://github.com/llvm-mirror/llvm/blob/HEAD/lib/Support/Host.cpp
         # and https://en.wikipedia.org/wiki/List_of_Intel_CPU_microarchitectures#Roadmap
+        vendor_id = cpuinfo[/^vendor_id\s*: (.*)/, 1]
         cpu_family = cpuinfo[/^cpu family\s*: ([0-9]+)/, 1].to_i
         cpu_model = cpuinfo[/^model\s*: ([0-9]+)/, 1].to_i
         unknown = :"unknown_0x#{cpu_family.to_s(16)}_0x#{cpu_model.to_s(16)}"
-        case cpu_family
+        case vendor_id
+        when "GenuineIntel"
+          intel_family(cpu_family, cpu_model) || unknown
+        when "AuthenticAMD"
+          amd_family(cpu_family) || unknown
+        else
+          unknown
+        end
+      end
+
+      def intel_family(family, model)
```

Comment:
```suggestion
      def intel_family(family, cpu_model)
```

just to shrink the diff a bit